applications/source/cdc-debezium-source2/src/test/java/org/springframework/cloud/stream/app/source/cdc/CdcMySqlTestSupport.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.source.cdc;
+
+import java.time.Duration;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.testcontainers.containers.GenericContainer;
+
+import org.springframework.jdbc.core.JdbcTemplate;
+
+/**
+ * @author David Turanski
+ * @author Christian Tzolov
+ */
+
+@Tag("integration")
+public abstract class CdcMySqlTestSupport {
+
+	static final String DATABASE_NAME = "inventory";
+
+	static String MAPPED_PORT;
+
+	static GenericContainer debeziumMySQL = new GenericContainer<>("debezium/example-mysql:2.1.4.Final")
+		.withEnv("MYSQL_ROOT_PASSWORD", "debezium")
+		.withEnv("MYSQL_USER", "mysqluser")
+		.withEnv("MYSQL_PASSWORD", "mysqlpw")
+		// .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("mysql")))
+		.withExposedPorts(3306)
+		.withStartupTimeout(Duration.ofSeconds(120))
+		.withStartupAttempts(3);
+
+	static {
+		debeziumMySQL.start();
+	}
+
+	static JdbcTemplate jdbcTemplate;
+
+	@BeforeAll
+	static void setup() {
+		MAPPED_PORT = String.valueOf(debeziumMySQL.getMappedPort(3306));
+		jdbcTemplate = CdcTestUtils.jdbcTemplate(
+				"com.mysql.cj.jdbc.Driver",
+				"jdbc:mysql://localhost:" + MAPPED_PORT + "/" + DATABASE_NAME + "?enabledTLSProtocols=TLSv1.2",
+				"root",
+				"debezium");
```

Comment:
I think all of these can map to auto-configuration properties.
The `@DynamicPropertySource` is a good way to do that from Testcontainers data.

Reply:
Not sure if the @DynamicPropertySource would be useful here as we create the container once per test. 
IMO, this feature would be useful if the same container is shared amongst multiple tests?

Reply:
Not sure what you mean with "once per test" because I see `static GenericContainer debeziumMySQL`.
So, it is really created for all the tests inheriting `CdcMySqlTestSupport`.
That `@DynamicPropertySource` can be used from a `static` initializer as well, e.g.:
```
	@DynamicPropertySource
	static void mongoDbProperties(DynamicPropertyRegistry registry) {
		registry.add("spring.data.mongodb.uri", MongoDbTestContainerSupport::mongoDbUri);
		registry.add("spring.data.mongodb.database", () -> "test");
	}
```

but I still see your point about responsibility concern: this class is not Spring-aware, so probably it is OK to leave as is.